modules/core/src/main/java/org/apache/ignite/internal/processors/cache/persistence/wal/FileWriteAheadLogManager.java

```diff
@@ -2185,6 +2185,13 @@ private void initAlreadyCompressedSegments() {
 
             if (alreadyCompressed.length > 0)
                 segmentAware.onSegmentCompressed(alreadyCompressed[alreadyCompressed.length - 1].idx());
+            else {
+                // We have to set a starting index for the compressor.
+                FileDescriptor[] alreadyArchived = scan(walArchiveDir.listFiles(WAL_SEGMENT_FILE_FILTER));
```

Comment:
If wal history is quite big and there are many archived files the scan can be quite slow